src/borg/repository.py

```diff
@@ -1183,7 +1183,7 @@ def is_committed_segment(self, segment):
         while True:
             try:
                 tag, key, offset, _ = next(iterator)
-            except IntegrityError:
+            except IntegrityError:  # KILLING? because found to be "not committed"?
                 return False
             except StopIteration:
                 break
```

Comment:
Yes, these two are an inherent (though I think not very important) issue in a transacted store; it's difficult to tell an aborted transaction (which can of course have partial writes and all that) from a corrupted completed transaction.

This is mitigated in two ways:
- 1.0: only one commit is left alive, and if it is corrupted it will try all segments, failing its way through all data, until the first segment, then abort and tell to run check (which will do the same and in the end stick a commit on it).
- 1.1: commit tags generally survive, so if one is corrupted the previous transaction would be accepted, depending on what exactly was done with some data loss. For compaction this is further mitigated because each segment ends with a commit.